src/app/core/services/user/user.service.ts

```diff
@@ -209,4 +230,8 @@ export class UserService {
     };
     return this.apiService.post('security/password', data, options);
   }
+
+  private getResource(isBusinessCustomer: boolean): 'customers' | 'privatecustomers' {
+    return this.isAppTypeREST && !isBusinessCustomer ? 'privatecustomers' : 'customers';
+  }
```

Comment:
doesn't this duplicate some stuff from the selector output?

Reply:
No, the selector uses the logged in user to determine the rest resource, that's why for the user registration it is not possible to use the selector

Reply:
But the surrounding logic can be extracted as it is duplicated. A helper function would be needed that spits out the endpoint depending on a boolean flag.

Reply:
yip, like that! 😄 👍